[bug] Flaky `TestPage/minID,_maxID_and_limit_set` test on x86_64-linux since 0.13.0

### Describe the bug with a clear and concise description of what the bug is.

The test suite should reliably produce either passing or failing results.

### What's your GoToSocial Version?

0.13.3

### GoToSocial Arch

x86_64-linux

### What happened?

The `TestPage/minID,_maxID_and_limit_set` test fails sometimes like this:
```
ok      github.com/superseriousbusiness/gotosocial/internal/ap  0.062s
ok      github.com/superseriousbusiness/gotosocial/internal/api/activitypub/emoji       1.027s
ok      github.com/superseriousbusiness/gotosocial/internal/api/activitypub/users       16.550s
ok      github.com/superseriousbusiness/gotosocial/internal/api/auth    5.695s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/accounts 114.605s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/admin    69.462s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/bookmarks        4.053s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/favourites       1.445s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/followrequests   24.882s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/instance 37.062s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/lists    9.235s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/media    8.924s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/polls    7.282s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/reports  44.547s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/search   27.412s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/statuses 26.424s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/streaming        0.901s
ok      github.com/superseriousbusiness/gotosocial/internal/api/client/user     3.179s
ok      github.com/superseriousbusiness/gotosocial/internal/api/fileserver      1.211s
ok      github.com/superseriousbusiness/gotosocial/internal/api/util    0.014s
ok      github.com/superseriousbusiness/gotosocial/internal/api/wellknown/webfinger     4.319s
ok      github.com/superseriousbusiness/gotosocial/internal/cache/domain        0.003s
ok      github.com/superseriousbusiness/gotosocial/internal/cleaner     5.482s
ok      github.com/superseriousbusiness/gotosocial/internal/config      0.165s
ok      github.com/superseriousbusiness/gotosocial/internal/db/bundb    94.391s
ok      github.com/superseriousbusiness/gotosocial/internal/email       0.178s
ok      github.com/superseriousbusiness/gotosocial/internal/federation  61.009s
ok      github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing    43.850s
ok      github.com/superseriousbusiness/gotosocial/internal/federation/federatingdb     2.518s
ok      github.com/superseriousbusiness/gotosocial/internal/gtserror    0.039s
ok      github.com/superseriousbusiness/gotosocial/internal/httpclient  0.082s
ok      github.com/superseriousbusiness/gotosocial/internal/language    0.030s
ok      github.com/superseriousbusiness/gotosocial/internal/log 0.207s
ok      github.com/superseriousbusiness/gotosocial/internal/media       49.926s
ok      github.com/superseriousbusiness/gotosocial/internal/middleware  0.209s
ok      github.com/superseriousbusiness/gotosocial/internal/oauth       1.144s
--- FAIL: TestPage (0.00s)
    --- FAIL: TestPage/minID,_maxID_and_limit_set (0.00s)
        page_test.go:67: minID, maxID and limit set
            page=&{Min:{Name:min_id Value:01HPVRBMDJ6EXFFH6TDEA2SMBH Order:Ascending} Max:{Name:max_id Value:01HPVRB1VT9V6ZHT837W7NNW8C Order:Descending} Limit:12} input=[01HPVR9B5TMRJWG67RMMKCR2EE 01HPVR9C52C3VAY1QKT3GHHK29 01HPVR9D4AXDQNZHAZBDF6588Z 01HPVR9E3JJVPYXC813XXS9YYX 01HPVR9F2TVW719EWTE41MK7CG 01HPVR9G22MCGTWA0J0NP9DEF5 01HPVR9H1AF873TBERFGRW2GKY 01HPVR9J0JCH621E8P1W121FAD 01HPVR9JZTZGRAXH95BJYSEZV7 01HPVR9KZ2SXTZ6FYYXNSB0B91 01HPVR9MYA50G8MJPY4WFGPWM8 01HPVR9NXJ4ES714SRFKRTEV4K 01HPVR9PWT23RWMNP3J7ZQ591A 01HPVR9QW27ZFG7E8C7CQFNA6N 01HPVR9RVAT6P2EEP5T4S2N4R6 01HPVR9STJX1FNXM3NFQ76W623 01HPVR9TSTRC4KQDXMM75T6HM4 01HPVR9VS2XRWC6A7ZDTC78BQK 01HPVR9WRA70ZKGZNAENTZ4V5D 01HPVR9XQJ2E5PFA4D49NC8XTX 01HPVR9YPTVEFS7PMTY0ASE2C4 01HPVR9ZP2HD65GMRPWCNKEMRX 01HPVRA0NAXNDYFAE0QT4N8YRX 01HPVRA1MJ9229GRWFJPT9EG9W 01HPVRA2KTKQ79Q6BD2D4B755M 01HPVRA3K2XEVQPT0N90NTS8DX 01HPVRA4JAEWXSYGYKHXA3NP66 01HPVRA5HJ3DV7JV1AFJJ4N1DP 01HPVRA6GTXG7T1BBFWT45P72F 01HPVRA7G20RAER4JZEQPTPHF3 01HPVRA8FAWEMB07HJCX0YDMCZ 01HPVRA9EJYSTK9BBTWD9G1KG3 01HPVRAADTTPC3K1GNXCW0A0B9 01HPVRABD2612Y7CJ82N66TN94 01HPVRACCAAGJJY3J78HXWXB4B 01HPVRADBJPEQYDHZ861WX50MR 01HPVRAEAT2HS9WFQBM1016WCX 01HPVRAFA2GY17V6QE9A4BEEHZ 01HPVRAG9ATCSD541BFPBJS1X8 01HPVRAH8JPK9FVATJD6GVPW01 01HPVRAJ7T3WC0CHSR8MM0H08T 01HPVRAK72MASZR229FZNWKVEK 01HPVRAM6AYW945K6JCH3YM5ZF 01HPVRAN5JPF8GBMYRF5V6XWC2 01HPVRAP4TN74WRN9FD0TSXSAB 01HPVRAQ42AK3B85FF8KRQWRVF 01HPVRAR3AF2BH0HSWV0MCX4J4 01HPVRAS2JJ11DK20KRHNESZBN 01HPVRAT1TG3BPZQTBXYTQATHM 01HPVRAV122Z7CTT7FW11RW4B6 01HPVRAW0A418K4X0W93H4S5XD 01HPVRAWZJN36NCG84RXHCFNSJ 01HPVRAXYTRHTSC84ZY6KW8EPR 01HPVRAYY2C7HG7M32PBXCCA8M 01HPVRAZXA0237WMA8VWK2E94K 01HPVRB0WJ7TBVCT2N2T1B9PPS 01HPVRB1VT9V6ZHT837W7NNW8C 01HPVRB2V2SKQS8R9VRPEH038W 01HPVRB3TA5CNM140CBR6CZX64 01HPVRB4SJ0V5ZHSG2KWT0K3FH 01HPVRB5RTA396MW5YF5NEQBYV 01HPVRB6R2VRGDN521QK85JCV4 01HPVRB7QAW8EQ96PGJFKQ4JQY 01HPVRB8PJDVZ59S9DD7WQJNAH 01HPVRB9NT3QKMTWAXZSM9GC85 01HPVRBAN2XJ805REERSN2HQDY 01HPVRBBMAY5CM73HTRSY8QXSP 01HPVRBCKJRYTMMVKT6T4V9G7A 01HPVRBDJTT973PB5T00753JYV 01HPVRBEJ2J6JB4JYNKEK68EPZ 01HPVRBFHAZ6N1VEQ0AS1VTPX5 01HPVRBGGJNKZKQ7VXFQ9B2FZ2 01HPVRBHFTE43RBEG9M7N7Z6NB 01HPVRBJF241VF93W3V9800XAA 01HPVRBKEA56XT2B5GXGH5BEPK 01HPVRBMDJ6EXFFH6TDEA2SMBH 01HPVRBNCTXP8TSXPGTDR6ZQP9 01HPVRBPC2AC0YFXA4H01Q3TB7 01HPVRBQBAQTN52R93CVAVTT2N 01HPVRBRAJGW7D4NABGBFCHDEN 01HPVRBS9TQB9H5HVFZ3JHFQSA 01HPVRBT92697P9KZA2M841FNX 01HPVRBV8A29S0Q4J2GWXP3R05 01HPVRBW7J5AWEN6X360B5NMWM 01HPVRBX6TNCJEC1DYFR870H8J 01HPVRBY62EA6Q01H6SEKKMC2A 01HPVRBZ5AVA9T4YJ14F1J2DPV 01HPVRC04JC4XJA11MGB867TGJ 01HPVRC13TGQ2SMA8E9TR5K6S2] expect=[01HPVRC13TGQ2SMA8E9TR5K6S2 01HPVRC04JC4XJA11MGB867TGJ 01HPVRBZ5AVA9T4YJ14F1J2DPV 01HPVRBY62EA6Q01H6SEKKMC2A 01HPVRBX6TNCJEC1DYFR870H8J 01HPVRBW7J5AWEN6X360B5NMWM 01HPVRBV8A29S0Q4J2GWXP3R05 01HPVRBT92697P9KZA2M841FNX 01HPVRBS9TQB9H5HVFZ3JHFQSA 01HPVRBRAJGW7D4NABGBFCHDEN 01HPVRBQBAQTN52R93CVAVTT2N 01HPVRBPC2AC0YFXA4H01Q3TB7] output=[01HPVRC04JC4XJA11MGB867TGJ 01HPVRBZ5AVA9T4YJ14F1J2DPV 01HPVRBY62EA6Q01H6SEKKMC2A 01HPVRBX6TNCJEC1DYFR870H8J 01HPVRBW7J5AWEN6X360B5NMWM 01HPVRBV8A29S0Q4J2GWXP3R05 01HPVRBT92697P9KZA2M841FNX 01HPVRBS9TQB9H5HVFZ3JHFQSA 01HPVRBRAJGW7D4NABGBFCHDEN 01HPVRBQBAQTN52R93CVAVTT2N 01HPVRBPC2AC0YFXA4H01Q3TB7 01HPVRBNCTXP8TSXPGTDR6ZQP9]
        page_test.go:71: unexpected paged output
FAIL
FAIL    github.com/superseriousbusiness/gotosocial/internal/paging      0.157s
FAIL
```

### What you expected to happen?

The `TestPage/minID,_maxID_and_limit_set` test always pass.

### How to reproduce it?

Run the test suite until it fails.

### Anything else we need to know?

_No response_

--- a/internal/paging/page_test.go
+++ b/internal/paging/page_test.go
@@ -20,15 +20,13 @@ package paging_test
 import (
 	"math/rand"
 	"slices"
+	"strconv"
 	"testing"
 	"time"
 
-	"github.com/oklog/ulid"
+	"github.com/stretchr/testify/assert"
 	"github.com/superseriousbusiness/gotosocial/internal/paging"
 )
-
-// random reader according to current-time source seed.
-var randRd = rand.New(rand.NewSource(time.Now().Unix()))
 
 type Case struct {
 	// Name is the test case name.
@@ -63,13 +61,9 @@ func TestPage(t *testing.T) {
 			// Page the input slice.
 			out := c.Page.Page(c.Input)
 
-			// Log the results for case of error returns.
-			t.Logf("%s\npage=%+v input=%v expect=%v output=%v", c.Name, c.Page, c.Input, c.Expect, out)
-
-			// Check paged output is as expected.
-			if !slices.Equal(out, c.Expect) {
-				t.Error("unexpected paged output")
-			}
+			// Check paged output is expected.
+			assert.Equal(t, c.Expect, out,
+				"input=%#v page=%v", c.Input, c.Page)
 		})
 	}
 }
@@ -80,8 +74,7 @@ var cases = []Case{
 		slices.SortFunc(ids, ascending)
 
 		// Select random indices in slice.
-		minIdx := randRd.Intn(len(ids))
-		maxIdx := randRd.Intn(len(ids))
+		minIdx, maxIdx, _ := generateParams(len(ids))
 
 		// Select the boundaries.
 		minID := ids[minIdx]
@@ -104,9 +97,7 @@ var cases = []Case{
 		slices.SortFunc(ids, ascending)
 
 		// Select random parameters in slice.
-		minIdx := randRd.Intn(len(ids))
-		maxIdx := randRd.Intn(len(ids))
-		limit := randRd.Intn(len(ids)) + 1
+		minIdx, maxIdx, limit := generateParams(len(ids))
 
 		// Select the boundaries.
 		minID := ids[minIdx]
@@ -116,12 +107,10 @@ var cases = []Case{
 		expect := slices.Clone(ids)
 		expect = cutLower(expect, minID)
 		expect = cutUpper(expect, maxID)
-		slices.Reverse(expect)
-
-		// Now limit the slice.
 		if limit < len(expect) {
 			expect = expect[:limit]
 		}
+		slices.Reverse(expect)
 
 		// Return page and expected IDs.
 		return ids, &paging.Page{
@@ -135,8 +124,7 @@ var cases = []Case{
 		slices.SortFunc(ids, ascending)
 
 		// Select random parameters in slice.
-		minIdx := randRd.Intn(len(ids))
-		maxIdx := randRd.Intn(len(ids))
+		minIdx, maxIdx, _ := generateParams(len(ids))
 
 		// Select the boundaries.
 		minID := ids[minIdx]
@@ -160,8 +148,7 @@ var cases = []Case{
 		slices.SortFunc(ids, descending)
 
 		// Select random indices in slice.
-		sinceIdx := randRd.Intn(len(ids))
-		maxIdx := randRd.Intn(len(ids))
+		sinceIdx, maxIdx, _ := generateParams(len(ids))
 
 		// Select the boundaries.
 		sinceID := ids[sinceIdx]
@@ -183,7 +170,7 @@ var cases = []Case{
 		slices.SortFunc(ids, descending)
 
 		// Select random indices in slice.
-		maxIdx := randRd.Intn(len(ids))
+		_, maxIdx, _ := generateParams(len(ids))
 
 		// Select the boundaries.
 		maxID := ids[maxIdx]
@@ -202,7 +189,7 @@ var cases = []Case{
 		slices.SortFunc(ids, descending)
 
 		// Select random indices in slice.
-		sinceIdx := randRd.Intn(len(ids))
+		sinceIdx, _, _ := generateParams(len(ids))
 
 		// Select the boundaries.
 		sinceID := ids[sinceIdx]
@@ -221,7 +208,7 @@ var cases = []Case{
 		slices.SortFunc(ids, ascending)
 
 		// Select random indices in slice.
-		minIdx := randRd.Intn(len(ids))
+		minIdx, _, _ := generateParams(len(ids))
 
 		// Select the boundaries.
 		minID := ids[minIdx]
@@ -258,32 +245,34 @@ func cutUpper(in []string, bound string) []string {
 	return in
 }
 
+// random reader according to current-time source seed.
+var randRd = rand.New(rand.NewSource(time.Now().Unix()))
+
+// generateParams ...
+func generateParams(n int) (minIdx int, maxIdx int, limit int) {
+	maxIdx = max(1, randRd.Intn(n))
+	minIdx = randRd.Intn(maxIdx)
+	limit = randRd.Intn(max(1, maxIdx-minIdx)) + 1
+	return
+}
+
 // generateSlice generates a new slice of len containing ascending sorted slice.
 func generateSlice(len int) []string {
-	if len <= 0 {
+	if len <= 1 {
 		// minimum testable
 		// pageable amount
 		len = 2
 	}
-	now := time.Now()
 	in := make([]string, len)
 	for i := 0; i < len; i++ {
-		// Convert now to timestamp.
-		t := ulid.Timestamp(now)
-
-		// Create anew ulid for now.
-		u := ulid.MustNew(t, randRd)
-
-		// Add to slice.
-		in[i] = u.String()
-
-		// Bump now by 1 second.
-		now = now.Add(time.Second)
+		in[i] = strconv.Itoa(i)
 	}
 	return in
 }
 
-func ascending(a, b string) int {
+func ascending(sa, sb string) int {
+	a, _ := strconv.Atoi(sa)
+	b, _ := strconv.Atoi(sb)
 	if a > b {
 		return 1
 	} else if a < b {
@@ -292,7 +281,9 @@ func ascending(a, b string) int {
 	return 0
 }
 
-func descending(a, b string) int {
+func descending(sa, sb string) int {
+	a, _ := strconv.Atoi(sa)
+	b, _ := strconv.Atoi(sb)
 	if a < b {
 		return 1
 	} else if a > b {

